config/routes.rb

```diff
@@ -172,6 +172,9 @@ def redirect(path, options = { prefix: Whitehall.router_prefix })
         get :edit_slug, on: :member, controller: :edition_slug
         patch :update_slug, on: :member, controller: :edition_slug
 
+        get :edit_change_note, on: :member, controller: :draft_edition_change_notes
+        patch :update_change_note, on: :member, controller: :draft_edition_change_notes
```

Comment:
There's a pretty well established of using resources where possible. I've used custom routes here for 2 reasons:

1. There's already a change notes controller that handles editing published change notes. Only GDS Admins have access to this. But currently, it's using the `editions/:id/change_notes` endpoints.

If we proceed with this i'd suggest we rename that controller to something that conveys it's use case.

2. the other reason custom endpoints may make sense to use though, is when you create rails resources by default they assume an id exists for the object, which in the case of form objects it rarely does.

We could potentially use shallow nesting https://guides.rubyonrails.org/routing.html#shallow-nesting instead.

I'm don't feel particularly strong about deviating from using resources, but keen to hear other peoples opinions. There could also def be something fairly simple that i'm missing that would allow us to build the resources without it adding an additional id param.

Reply:
Hmm, I think I agree with your choice. Using shallow nesting would be going against its intended use of nesting index and create endpoints for a resource. Do [singular resources](https://guides.rubyonrails.org/routing.html#singular-resources) work in this case? I'm guessing having create and delete routes for change notes doesn't really make much sense?

Reply:
Yea good shout i think removing the `on: :member` makes sense here so that it expects :edition_id rather than just :id.

Yea i think the majority of the resources aren't required for this since it's just an edit page. 

I've ended up going with

```
 get :edit_change_note, to: 'draft_edition_change_notes#edit'
 patch :update_change_note, to: 'draft_edition_change_notes#update'
```

So the action names can be generic. it required quite a lot of noisy changes (file name changes etc.) so i just squashed it into the second commit. Let me know what you think